### PR TITLE
Rename JWebClient to JApplicationWebClient for autoloading.

### PR DESCRIPTION
**Pull request by realityking**
Also add deprecation warnings to JRule and JRules.

### CODE REVIEW

**Comment by chdemko** [comment]
@realityking tests/suite/joomla/application/JApplicationWebTest.php:150

Change JWebClient to JApplicationWebClient

File: tests/suite/joomla/application/web/JApplicationWebClientTest.php [hunk 1]
```diff
@@ -0,0 +1,254 @@
+<?php
+/**
+ * @package     Joomla.UnitTest
+ * @subpackage  Application
+ *
+ * @copyright   Copyright (C) 2005 - 2012 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE
+ */
+
+require_once JPATH_PLATFORM.'/joomla/application/web/webclient.php';
```
Comment (chdemko) [inline tests/suite/joomla/application/web/JApplicationWebClientTest.php]: This should be

```
require_once JPATH_PLATFORM.'/joomla/application/web/client.php';
```

File: tests/suite/joomla/application/web/JApplicationWebClientTest.php [hunk 2]
```diff
@@ -0,0 +1,254 @@
+<?php
+/**
+ * @package     Joomla.UnitTest
+ * @subpackage  Application
+ *
+ * @copyright   Copyright (C) 2005 - 2012 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE
+ */
+
+require_once JPATH_PLATFORM.'/joomla/application/web/lient.php';
```
Comment (chdemko) [inline tests/suite/joomla/application/web/JApplicationWebClientTest.php]: you miss a 'c' ('c'lient)

Comment (realityking) [inline tests/suite/joomla/application/web/JApplicationWebClientTest.php]: Gosh, darn. Fixed. Thanks.

**Pull request merged**